source/CMakeLists.txt

```diff
@@ -5,11 +5,22 @@ add_library(asl ${SRC_LIST} ${INC_LIST})
 target_include_directories(asl PUBLIC ..)
 target_compile_definitions(asl PUBLIC ADOBE_STD_SERIALIZATION)
 
-#otherwise boost concept checking fails on adobe::selection_t because it lacks non const begin/end
+# otherwise boost concept checking fails on adobe::selection_t because it lacks non const begin/end
 target_compile_definitions(asl PUBLIC -DBOOST_RANGE_ENABLE_CONCEPT_ASSERT=0)
 
-#otherwise boost::move and std::move are ambiguous
-target_compile_definitions(asl PUBLIC -DBOOST_MOVE_USE_STANDARD_LIBRARY_MOVE)
+# otherwise boost::move and std::move are ambiguous
+# commented out here because ADOBE_STD_ flags below make it irrelevant
+#target_compile_definitions(asl PUBLIC BOOST_MOVE_USE_STANDARD_LIBRARY_MOVE)
+
+if (USE_STD_THREAD_LOCAL)
+    target_compile_definitions(asl PUBLIC ADOBE_STD_THREAD_LOCAL)
+endif()
+
+# GCC6 warning on boost 1.60
+# right operand of shift expression '(1u << 63u)' is >= than the precision of the left operand [-fpermissive]
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
+    target_compile_definitions(asl PUBLIC ADOBE_FNV_NO_BIGINTS)
```

Comment:
admittedly, this is quite an aggressive "fix" to silence a warning ... I just don't need big int.
Seems boost 1.61 adressed this issue by the way
https://github.com/boostorg/multiprecision/commit/f9c8f9ec091ad232c0a291904f7839d665d098e0